File: common/src/main/java/com/hartwig/actin/clinical/datamodel/Treatment.java
package com.hartwig.actin.clinical.datamodel;

import java.util.Set;

import org.jetbrains.annotations.NotNull;

public interface Treatment {

    @NotNull
    String name();

    @NotNull
    Set<TreatmentCategory> categories();

    @NotNull
    Set<String> synonyms();

    boolean isSystemic();
}

File: soc/src/main/kotlin/com/hartwig/actin/soc/datamodel/Treatment.kt
package com.hartwig.actin.soc.datamodel

import com.hartwig.actin.clinical.datamodel.TreatmentCategory
import com.hartwig.actin.treatment.datamodel.EligibilityFunction

data class Treatment(
    val name: String,
    val isOptional: Boolean,
    val score: Int,
    val components: Set<TreatmentComponent>,
    val categories: Set<TreatmentCategory>,
    val eligibilityFunctions: Set<EligibilityFunction>,
    val lines: Set<Int>
)